### ISSUE TITLE
[BUG] Aggregation object's destructor is not virtual

### ISSUE THREAD

**Issue by devavret** [rapidsai/cudf#4535]
**Describe the bug**
`aggregation` is a parent class to other aggregation objects that can take more arguments, however the factories return `std::unique_ptr<aggregation>`. This destructor should be virtual: https://github.com/rapidsai/cudf/blob/9c429e9151f3a153da555082a878da32046ce4a5/cpp/include/cudf/aggregation.hpp#L77

**Steps/Code to reproduce bug**
```c++
TYPED_TEST(groupby_quantile_test, leak)
{
    for (size_t i = 0; i < 100; i++) {
        auto agg = cudf::experimental::make_quantile_aggregation({0});
    }
}
```
`GTEST_FILTER="*leak*" valgrind ./cudf/cpp/build/gtests/GROUPBY_TEST`
```log
==28854== LEAK SUMMARY:
==28854==    definitely lost: 4,800 bytes in 600 blocks
```

**Comment by jrhemstad** [comment]
See, happens to the best of us :)

**Comment by devavret** [comment]
> See, happens to the best of us :)

I thought this was again me. Was scrambling to fix all place I remember making this mistake.

**Comment by devavret** [comment]
Closed by #4617
